### PR TITLE
fix(curriculum): rework Project Euler 89

### PR DESCRIPTION
**Pull request by gikf**
- Adds second test.
- Adds solution. 
- Defines arrays with tests numerals in background.
- Removes external link in favor of expanded description. Per #40749.
- Partially related to #40896.
- Tested on local fork.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md [hunk 1]
```diff
@@ -8,7 +8,18 @@ dashedName: problem-89-roman-numerals
 
 # --description--
 
-For a number written in Roman numerals to be considered valid there are basic rules which must be followed. Even though the rules allow some numbers to be expressed in more than one way there is always a "best" way of writing a particular number.
+For a number written in Roman numerals to be considered valid there are basic rules which must be followed. Even though the rules allow some numbers to be expressed in more than one way there is always a *best* way of writing a particular number.
+
+- Numerals must be arranged in descending order of size.
+- M, C, and X cannot be equalled or exceeded by smaller denominations.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:14]: This is a non-American way of spelling.

```suggestion
- M, C, and X cannot be equaled or exceeded by smaller denominations.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md [hunk 2]
```diff
@@ -18,52 +29,89 @@ For example, it would appear that there are at least six ways of writing the num
...
   XIIIIII<br>
   VVVI<br>
-  XVI<br>
+  XVI<br><br>
 </div>
 
 However, according to the rules only XIIIIII and XVI are valid, and the last example is considered to be the most efficient, as it uses the least number of numerals.
 
-The array, `roman`, contains one thousand numbers written in valid, but not necessarily minimal, Roman numerals; see [About... Roman Numerals](https://projecteuler.net/about=roman_numerals) for the definitive rules for this problem.
+The array, `roman`, will contain numbers written in valid, but not necessarily minimal, Roman numerals.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:37]:
```suggestion
The array, `roman`, will contain numbers written invalid, but not necessarily minimal, Roman numerals.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md (hunk 2 above)
Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:37]: It's supposed to be array with valid numerals, so invalid definitely cannot be there. Replace it with something else? *Correctly*? Or leave as it is?

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md (hunk 1 above)
Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:14]: Looking around both seems to be used, one is preferred in the American-, another in British-English. Not sure if there's one that's preferred here.

**Review by Sembauke — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md (hunk 1 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:14]: We tend to follow American spellings, I believe.

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md (hunk 2 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md:37]: Maybe "written with valid"

**Review by Sembauke — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Comment by moT01** [comment]
I added this solution to the forum guide article

**Pull request merged**